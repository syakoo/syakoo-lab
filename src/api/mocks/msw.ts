import { handleAPIFetchArtFav } from "@/api/fav/arts/fetch-art-fav.mocks";
import { handleAPIIncrementArtFav } from "@/api/fav/arts/increment-art-fav.mocks";
import { handleAPISyncArtsFav } from "@/api/fav/arts/sync-arts-fav.mocks";

export const defaultHandlers = [
  handleAPIFetchArtFav,
  handleAPIIncrementArtFav,
  handleAPISyncArtsFav,
];
