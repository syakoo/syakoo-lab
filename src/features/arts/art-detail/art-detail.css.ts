import { style } from "@vanilla-extract/css";

import { theme } from "@/design-system/theme.css";

export const artDetailStyles = {
  root: style({
    paddingBlock: theme.space[400],
  }),
  heartButtonWrapper: style({
    width: 36,
  }),
  imageWrapper: style({
    width: "100%",
    maxWidth: theme.size[100],
    margin: "auto",
  }),
  image: style({
    width: "100%",
    height: "auto",
  }),
  body: style({
    maxWidth: "500px",
    margin: "auto",
    marginTop: theme.space[300],
    paddingInline: theme.space[200],
    width: "100%",
    display: "flex",
    flexDirection: "column",
    gap: theme.space[100],
  }),
};
