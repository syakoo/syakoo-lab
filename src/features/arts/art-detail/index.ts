export { ArtDetail } from "./art-detail";
