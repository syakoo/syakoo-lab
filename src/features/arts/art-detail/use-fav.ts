import { useState } from "react";

import { api } from "@/api/reaction-system";
import { convertArtIdToReactionId } from "@/features/arts/_shared/art-id-to-reaction-id";
import { useMount } from "@/utils/mount/use-mount";

type UseFavReturn = {
  fav: number | null;
  incrementFav: () => Promise<void>;
};

export const useFav = (id: string): UseFavReturn => {
  const [fav, setFav] = useState<number | null>(null);
  // 更新分の値
  const [incrementedValue, setIncrementedValue] = useState<number>(0);

  useMount(() => {
    void (async () => {
      const result = await api.getReaction(convertArtIdToReactionId(id));
      setFav(result.likes);
    })();
  });

  const incrementFav = async () => {
    setIncrementedValue(1);
    await api.postReactionIncrement(convertArtIdToReactionId(id), {
      likes: 1,
    });
  };

  const resolvedFav = (() => {
    if (fav === null) return null;
    return fav + incrementedValue;
  })();

  return { fav: resolvedFav, incrementFav };
};
