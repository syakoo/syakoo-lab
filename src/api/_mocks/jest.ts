import type { RequestHandler } from "msw";
import { setupServer } from "msw/node";

/**
 * jest 用のモックサーバー起動関数
 */
export function setupJestMockServer(...handlers: RequestHandler[]) {
  const server = setupServer(...handlers);
  beforeAll(() => server.listen());
  afterEach(() => server.resetHandlers());
  afterAll(() => server.close());
  return server;
}
