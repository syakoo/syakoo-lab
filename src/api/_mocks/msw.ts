import { defaultHandlers as favArtsDefaultHandlers } from "../fav/arts/_mocks/msw";

export const defaultHandlers = [...favArtsDefaultHandlers];
