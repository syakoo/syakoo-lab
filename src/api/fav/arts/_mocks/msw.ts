import { type HttpHandler, http, delay, HttpResponse } from "msw";

import {
  FetchArtFavResponseBody,
  fetchArtFavPath,
} from "@/api/fav/arts/fetchArtFav";
import { incrementArtFavPath } from "@/api/fav/arts/incrementArtFav";
import { syncArtsFavPath } from "@/api/fav/arts/syncArtsFav";

export const handleSyncArtsFav: HttpHandler = http.post(
  syncArtsFavPath(),
  async () => {
    await delay(1000);
    return new HttpResponse();
  },
);

export const handleFetchArtFav: HttpHandler = http.get(
  fetchArtFavPath(":id"),
  async () => {
    await delay(1000);
    return HttpResponse.json<FetchArtFavResponseBody>({ fav: 10 });
  },
);

export const handleIncrementArtFav: HttpHandler = http.post(
  incrementArtFavPath(":id"),
  async () => {
    await delay(1000);
    return HttpResponse.json<FetchArtFavResponseBody>({ fav: 11 });
  },
);

export const defaultHandlers: HttpHandler[] = [
  handleSyncArtsFav,
  handleFetchArtFav,
  handleIncrementArtFav,
];
