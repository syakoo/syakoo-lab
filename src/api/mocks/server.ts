import { createMiddleware } from "@mswjs/http-middleware";
import cors from "cors";
import express from "express";

import { supabaseUrl } from "@/api/supabase/_shared/config";

import { defaultHandlers } from "./msw";

/**
 * ローカル環境用のモックサーバー起動関数
 */
export const runServer = () => {
  const app = express();

  app.use(cors());
  app.use(express.json());
  app.use(createMiddleware(...defaultHandlers));
  app.use((_req, res) => {
    res.status(404).json({
      error: "Mock not found",
    });
  });

  const { port } = new URL(supabaseUrl);
  app.listen(port);
};
