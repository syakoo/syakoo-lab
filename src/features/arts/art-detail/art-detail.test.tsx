import { setupJestMockServer } from "@/api/mocks/jest";
import { defaultHandlers } from "@/api/mocks/msw";
import { testStories } from "@/test/test-stories";

import * as stories from "./art-detail.stories";

setupJestMockServer(...defaultHandlers);

describe("ArtDetail", () => {
  testStories(stories);
});
