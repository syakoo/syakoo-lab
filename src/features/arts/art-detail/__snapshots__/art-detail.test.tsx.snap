// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtDetail スナップショット Sample 1`] = `
<body>
  <div>
    <article
      class="art-detail_artDetailStyles_root__w0s1yq0"
    >
      <div
        class="art-detail_artDetailStyles_imageWrapper__w0s1yq2"
      >
        <img
          alt="Art Title"
          class="art-detail_artDetailStyles_image__w0s1yq3"
          data-nimg="1"
          decoding="async"
          height="1500"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fplacehold.jp%2F3063D4%2Fffffff%2F2000x1500.png&w=3840&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fplacehold.jp%2F3063D4%2Fffffff%2F2000x1500.png&w=3840&q=75 1x"
          style="color: transparent;"
          width="2500"
        />
      </div>
      <div
        class="art-detail_artDetailStyles_body__w0s1yq4"
      >
        <div
          class="flex_flexStyle__12xne02i flex_flexStyle_align_flexEnd__12xne02t flex_flexStyle_variants_align_flexEnd__12xne028 flex_flexStyle_direction_row__12xne02j flex_flexStyle_gap_50__12xne02x flex_flexStyle_variants_gap_50__12xne02c flex_flexStyle_justify_flexStart__12xne02l flex_flexStyle_variants_justify_flexStart__12xne020"
        >
          <div
            class="art-detail_artDetailStyles_heartButtonWrapper__w0s1yq1"
          >
            <button
              aria-label="heart button"
              class="heart-button_heartButtonStyles_root__gy39e23"
              type="button"
            >
              <div
                class="heart-button_heartButtonStyles_iconWrapper__gy39e24"
              >
                <svg
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 8.25c0-2.485-2.099-4.5-4.688-4.5-1.935 0-3.597 1.126-4.312 2.733-.715-1.607-2.377-2.733-4.313-2.733C5.1 3.75 3 5.765 3 8.25c0 7.22 9 12 9 12s9-4.78 9-12z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </div>
            </button>
          </div>
          <div>
             
          </div>
        </div>
        <div>
          <h2
            class="text_textStyle__xbx8xue text_textStyle_color_currentColor__xbx8xui text_textStyle_variants_color_currentColor__xbx8xu3 text_textStyle_size_400__xbx8xuo text_textStyle_variants_size_400__xbx8xu9 text_textStyle_weight_bold__xbx8xut"
          >
            Art Title
          </h2>
          <div
            class="flex_flexStyle__12xne02i flex_flexStyle_align_stretch__12xne02u flex_flexStyle_variants_align_stretch__12xne029 flex_flexStyle_direction_column__12xne02k flex_flexStyle_gap_100__12xne02y flex_flexStyle_variants_gap_100__12xne02d flex_flexStyle_justify_flexStart__12xne02l flex_flexStyle_variants_justify_flexStart__12xne020"
          >
            <div>
              <p>
                Sample MDX Component
              </p>
              <h2>
                Sample Title
              </h2>
            </div>
            <div>
              <div
                class="flex_flexStyle__12xne02i flex_flexStyle_align_stretch__12xne02u flex_flexStyle_variants_align_stretch__12xne029 flex_flexStyle_direction_row__12xne02j flex_flexStyle_gap_50__12xne02x flex_flexStyle_variants_gap_50__12xne02c flex_flexStyle_justify_flexStart__12xne02l flex_flexStyle_variants_justify_flexStart__12xne020"
              >
                <p
                  class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_50__xbx8xuj text_textStyle_variants_size_50__xbx8xu4 text_textStyle_weight_normal__xbx8xuu"
                >
                  #
                  雑記
                </p>
                <p
                  class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_50__xbx8xuj text_textStyle_variants_size_50__xbx8xu4 text_textStyle_weight_normal__xbx8xuu"
                >
                  #
                  Sample
                </p>
              </div>
              <p
                class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_50__xbx8xuj text_textStyle_variants_size_50__xbx8xu4 text_textStyle_weight_normal__xbx8xuu"
              >
                2023-10-31
              </p>
            </div>
          </div>
        </div>
      </div>
    </article>
  </div>
</body>
`;
